common/data/Glicko.lua

```diff
@@ -0,0 +1,336 @@
+        end
+        player2Results[#player2Results+1] = player1:score(matchOutcome)
+    end
+    
+    local updatedPlayer1 = player1:update(player1Results)
+    local updatedPlayer2 = player2:update(player2Results)
+
+    return updatedPlayer1, updatedPlayer2
+end
+
+-- Attaches a score to an opponent
+function Glicko2:score(score)
+	local new_g2 = self:copy()
+	
+	--lost: 0, win: 1, tie: 0.5
+	new_g2.Score = score or 0
+	
+	return new_g2
+end
+
+-- Function g as described in step 3
+local function g(RD)
+	return 1/sqrt(1 + 3*RD^2/math.pi^2)
+end
+
+-- Function E as described in step 3
+local function E(rating, opRating, opRD)
+	return 1/(1 + exp(-g(opRD)*(rating - opRating)))
+end
+
+-- Constructor for function f described in step 5
+local function makebigf(g2, v, delta)
+	local a = log(g2.Vol^2)
+	
+	return function(x)
+		local numer = exp(x)*(delta^2 - g2.RD^2 - v - exp(x)) --numerator
+		local denom = 2*(g2.RD^2 + v + exp(x))^2 --denominator
+		local endTerm = (x - a)/(Glicko2.Tau^2) --final term
+		
+		return numer/denom - endTerm
+	end
+end
+
+-- Updates a Glicko rating using the last set of matches
+function Glicko2:update(matches)
+	local g2 = self
+	local originalVersion = g2.Version
+
+	-- convert ratings to glicko2
+	if originalVersion == 1 then
+		g2 = g2:to2()
+	end
+	
+  local convertedMatches = {}
+	for i, match in ipairs(matches) do
+		if match.Version == 1 then
+			convertedMatches[i] = match:to2()
+		end
+	end
+
+	-- step 3: compute v
+	local v = 0
+	
+	for j, match in ipairs(convertedMatches) do
+		local EValue = E(g2.Rating, match.Rating, match.RD)
+
+		v = v + g(match.RD)^2*EValue*(1 - EValue)
+	end
+	
+	v = 1/v
+
+	-- step 4: compute delta
+	local delta = 0
+
+	for j, match in ipairs(convertedMatches) do
+		local EValue = E(g2.Rating, match.Rating, match.RD)
+
+		delta = delta + g(match.RD)*(match.Score - EValue)
+	end
+
+	delta = delta*v
+
+	-- step 5: find new volatility (iterative process)
+	local a = log(g2.Vol^2)
+
+	local bigf = makebigf(g2, v, delta)
+
+	-- step 5.2: find initial A and B values
+	local A = a
+	local B = 0
+
+	if delta^2 > g2.RD^2 + v then
+		B = log(delta^2 - g2.RD^2 - v)
+	else
+		--iterative process for solving B
+		local k = 1
+
+		while bigf(a - k*Glicko2.Tau) < 0 do
+			k = k + 1
+		end
+
+		B = a - k*Glicko2.Tau
+	end
+
+	-- step 5.3: compute values of bigf of A and B
+	local fA = bigf(A)
+	local fB = bigf(B)
+
+	-- step 5.4: iterates until A and B converge
+	while math.abs(B - A) > epsilon do
+		local C = A + (A - B)*fA/(fB - fA)
+		local fC = bigf(C)
+
+		if fC*fB <= 0 then
+			A = B
+			fA = fB
+		else
+			fA = fA/2
+		end
+
+		B = C
+		fB = fC
+	end
+
+	-- step 5.5: set new volatility
+	local newVol = g2.Vol
+
+	if #convertedMatches > 0 then
+		newVol = exp(A/2)
+	end
+
+	-- step 6: update the rating deviation to the new pre-rating period value
+	local ratingDeviation = sqrt(g2.RD^2 + newVol^2)
+	
+	-- Step 7: Update to the new rating
+
+	local newRD = 1/sqrt(1/ratingDeviation^2 + 1/v)
```

Comment:
The example ranking you generated shows players with an RD below 30.
In the paper on Glicko it is stated that

> One practical problem with the Glicko system is that when a player competes very frequently,
his/her rating stops changing appreciably which reflects that the RD is very small. This may
sometimes prevent a player’s rating from changing substantially when the player is truly
improving. I would therefore recommend that an RD never drop below a threshold value,
such as 30, so that ratings can change appreciably even in a relatively short time.

For the Panel Attack ladder I believe this to be highly relevant with a small number of players that always keep ranked on driving their RD low compared to everyone playing ranked only on a weekly or monthly basis.